### DISCUSSION TITLE
NixOS Compatibility

### DISCUSSION THREAD

**Discussion by Bakan0** [mylinuxforwork/dotfiles#266] [Q&A] [answered]
Greetings! I've enjoyed watching your YouTube videos, and I found you from NixOS' [Example configurations page](https://wiki.hyprland.org/Configuring/Example-configurations/#stephan-raabe-ml4w), but I do not see any straight-forward way to get your setup on NixOS. Any information would be greatly appreciated.

Perhaps the questions for Q&A here would be:
1) Is this designed to be easily deployable into NixOS?
2a) If so, are there instructions for NixOS somewhere?
2b) If not, is there a difficult-but-documented way to install it?

Thank you for your hard work :D

**Comment by mylinuxforwork** [comment, marked as answer]
@Bakan0 Thanks for your question. You can try this https://github.com/mylinuxforwork/dotfiles/wiki/Using-ML4W-Dotfiles-on-other-distributions

But you have to adapt the update script if needed. Would be great if you could share your results.

**Reply by Bakan0** [reply]
Ah perfect :smile:  Exactly an answer to my question! I hope you have a happy and successful week, and thank you again for all you do for the community.

Should I have success, I will return and report!

**Comment by Schuasda** [comment]
a install script or detailed guide for NixOs would be greatly appreciated :D
But in the meantime I did try it on my own, and i think I am very close, but i get an error about colors:
`Error parsing gradient rgb({color11.strip}): No colors?`

I can't really find anything about this error online, I am very new to Hyprland, so I don't even know where to start looking.
Do you have any ideas how i could fix this? Am I missing a package or font or something?

**Reply by dwilliam62** [reply]
Make sure that wallust is installed and you set a wallpaper.   The 'color11' is a variable that needs to have a valid color value.

**Reply by Schuasda** [reply]
i did already have pywal (which from my understanding should do the same thing?) and i didn't change any of the configuration and figured there probably is a default wallpaper already set, but might also not work, because of the issue I was having.

Well it turns out i was missing the wallpaper folder in my home directory, which contains all the wallpaper files being shipped with ml4w. So thank you @dwilliam62 for the hint, you pointed me to the right place to look.
Everything works fine now, now i need to get into actually learning how to configure all the stuff to maybe adjust things to my tase a bit :D
